### DISCUSSION TITLE
expectJsonMatch and includes Usage

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#221] [Q&A] [answered]
Hi, 

I have used includes with expectJsonMatch in javascript in this way :- 
```ts
expectJsonMatch(
                'payload[*].classes[*]',
                includes({
                    id: 'ea99500d-8d52-fc52-b5a5-10dcd1e9d2bd',
                    displayName: 'Windows Computer',
                    singleton: false
                })
            );
```

I understand that in docs of pactum it says to use with the properties . However, in the above way it works nicely and serves my purpose aswell.
In the above code, there is a JSON with array of JSON objects and the array length is huge. This way it helps me to find out exactly the json which I need to verified.

Now, when I am switching to typescript, this option does not work anymore.
Any suggestion how to handle this ?

Thanks,
Mihir

**Comment by ASaiAnudeep** [comment, marked as answer]
[includes](https://pactumjs.github.io/api/matching/includes.html) only accepts a string as input. `expectJsonLike` might suite better here.

**Reply by mihirsasmal** [reply]
I have tried `expectJsonLike` and other options as well, nothing works for this scenario. It's because it needs an exact path which I don't have and I am searching in entire array.
Includes supposed to take string as input but it works in javascript the way I have mentioned above or it was giving a false positive. 
I would like the include to work that way as well (the way I was using it in javascript), but not sure if everyone prefers that approach .

**Reply by ASaiAnudeep** [reply]
Can you provide us the sample response and your expectations on it. I will try to find out a better way to assert it.

**Reply by mihirsasmal** [reply]
Hi, apologies for replying late as I was on holiday. 
Here is the [link](https://github.com/mihirsasmal/files/blob/66498d4301621c1795168df6e8b1709684b90e91/IncludeTest.Json) to sample json file. I want to make sure that the below json is available in entire JSON file. My usage in JavaScript is as below:

```
.expectJsonMatch(
                'rows[*][*]',
                includes({
                    raw: 'Azure DevOps',
                    value: 'Azure DevOps',
                    formatted: 'Azure DevOps'
                })
            );

```

**Reply by ASaiAnudeep** [reply]
This works perfectly fine.

```js
await spec()
    .get(<url>)
    .expectJsonLike({
      rows: [
        [
          {
            raw: 'Azure DevOps',
            value: 'Azure DevOps',
            formatted: 'Azure DevOps'
          }
        ]
      ]
    })
```

**Reply by mihirsasmal** [reply]
Awesome 👍  This way it works. Thanks AsaiAnudeep.
